clients/ts/FunctionalTests/ts/HubConnectionTests.ts

```diff
@@ -544,6 +544,31 @@ describe("hubConnection", () => {
         }
     });
 
+    it("transport falls back from WebSockets to SSE or LongPolling", async (done) => {
+        // Replace Websockets with a function that just
+        // throws to force fallback.
+        const oldWebSocket = (window as any).WebSocket;
+        (window as any).WebSocket = () => {
+            throw new Error("Kick rocks");
+        };
+
+        const hubConnection = new HubConnection(TESTHUBENDPOINT_URL, {
+            logger: LogLevel.Trace,
+            protocol: new JsonHubProtocol(),
+        });
+
+        try {
+            await hubConnection.start();
+
+            // Make sure that we connect with SSE.
+            expect(await hubConnection.invoke("GetActiveTransportName")).toEqual("ServerSentEvents" || "LongPolling");
```

Comment:
I don't think this works, I think you'd have to do something like

```typescript
let transportName = await hubConnection.invoke("GetActiveTransportName");
expect(transportName === "ServerSentEvents" || transportName === "LongPolling").toBe(true)
```

Truthy and falsey (the weirdest concepts in JavaScript) strike again, because `x || y` means "if `x` is truthy, it is the value of the expression, otherwise `y` is". Any non-empty string is truthy, so `"ServerSentEvents" || "LongPolling" === "ServerSentEvents"`.

JavaScript is super weird and confusing... this stuff gets everyone confused.